lib/internal/per_context/domexception.js

```diff
@@ -49,12 +49,23 @@ const disusedNamesSet = new SafeSet()
   .add('ValidationError');
 
 class DOMException {
-  constructor(message = '', name = 'Error') {
+  constructor(message = '', options = 'Error') {
     ErrorCaptureStackTrace(this);
-    internalsMap.set(this, {
-      message: `${message}`,
-      name: `${name}`
-    });
+
+    if (options && typeof options === 'object') {
+      const { name, cause } = options;
+      internalsMap.set(this, {
+        message: `${message}`,
+        name: `${name}`
+      });
+
+      ObjectDefineProperty(this, 'cause', { __proto__: null, value: cause });
```

Comment:
For folks who are reviewing this... there's some background about this being an own property rather than a getter.

For intrinsic JavaScript errors, the spec declares that `cause` is added as an own property such that the following cases can be differentiated:

```js
const error1 = new Error('no cause');
console.log('cause' in error1);  // false
console.log(error1.cause); // undefined

const error2 = new Error('undefined cause', { cause: 'foo' });
console.log('cause' in error2);  // true
console.log(error2.cause);  // undefined
```

For the WHATWG definition, however, they appear to be intentionally diverging from the pattern established by the language spec and declaring `DOMException` `cause` using a getter. In talking this change over with @flakey5, we opted to go with the approach that aligns with the basic language spec and intrinsic errors.

Reply:
Also cc @legendecas